test/fixtures/poro.rb

```diff
@@ -241,6 +252,22 @@ def self.root_name
   has_one :blog, key: :site
 end
 
+LinkTagSerializer = Class.new(ActiveModel::Serializer) do
+  has_many :posts, links: true
+end
+
+LinkNoDataTagSerializer = Class.new(ActiveModel::Serializer) do
+  has_many :posts, links: true, data: false
+end
+
+LinkUrlTagSerializer = Class.new(ActiveModel::Serializer) do
+  has_many :posts, links: { related: 'http://test.host/post_tags' }, data: false
+end
+
+LinkProcTagSerializer = Class.new(ActiveModel::Serializer) do
+  has_many :posts, links: { related: -> (object, name) { "http://test.host/tags/#{object.id}/#{name}/all" } }, data: false
```

Comment:
> related: -> (object, name) { "http://test.host/tags/#{object.id}/#{name}/all" }

I believe what you want here is a uri_template. See https://github.com/hannesg/uri_template 

See e.g. https://developer.github.com/v3/#schema http://tools.ietf.org/html/rfc6570 see e.g. https://api.github.com/ (though not a JSON API, is an example of usage)